### PR TITLE
Proper Cleanup

### PR DESCRIPTION
**Pull request by Schneegans**
This ensures that the plugin can be properly reloaded at runtime.

### CODE REVIEW

File: src/Atmosphere.cpp [hunk 1]
```diff
@@ -52,17 +52,17 @@ Atmosphere::Atmosphere(std::shared_ptr<Plugin::Properties> const& pProperties,
...
-      mAtmosphereNode, static_cast<int>(cs::utils::DrawOrder::eAtmospheres));
+      mAtmosphereNode.get(), static_cast<int>(cs::utils::DrawOrder::eAtmospheres));
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
 Atmosphere::~Atmosphere() {
   VistaSceneGraph* pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
-  pSG->GetRoot()->DisconnectChild(mAtmosphereNode);
+  pSG->GetRoot()->DisconnectChild(mAtmosphereNode.get());
```
Comment (bernstein) [inline src/Atmosphere.cpp:65]:
```suggestion
  if (pSG) {
    pSG->GetRoot()->DisconnectChild(mAtmosphereNode.get());
  }
```

**Review by bernstein — CHANGES_REQUESTED** [review]

File: src/Atmosphere.cpp (hunk 1 above)
Comment (Schneegans) [reply src/Atmosphere.cpp:65]: Are there cases in which the SceneGraph des not exist? I think we never check this pointer...

Comment (bernstein) [reply src/Atmosphere.cpp:65]: Even though the pointer is probably not null, i would feel better if we always check raw pointers and in this case there are no additional costs in doing so. but it's just a suggestion.

Comment (Schneegans) [reply src/Atmosphere.cpp:65]: In a properly started application it cannot be null. If we start checking every pointer we get from Vista...

In fact you would have to do:
```c++
auto vs = GetVistaSystem();
if(vs)
  auto gm = vs->GetGraphicsManager();
  if (gm)
    sg = gm->GetSceneGraph();
```

... and we have bazillions of such calls  throughout the code base!

**Pull request merged**